### PR TITLE
Added clarification around event handlers

### LINKED ISSUE #18624: EventHandler is not a handler and neither is handler
The delegate, EventHandler is not a handler at all, in the sense of "a method that handles an event".  So I think there should be a paragraph near the top that helps the reader wrap their head around this:

_Microsoft has used the name EventHandler for the delegate name and this can be confusing for those new to this topic.  An instance of the EventHandler delegate creates/publishes the event and has the method that fires the event.  It should not be confused with the method that handles the event._

Also the example uses this line:

`   EventHandler<CustomEventArgs> handler = RaiseCustomEvent;`

Which piles on the confusion.  This is another case where the word handler is used for something that is not a handler.

Otherwise a very helpful example/tutorial.  I definitely notice MS improving in this area.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 38e8321f-5516-e451-5429-750d49f0ebed
* Version Independent ID: 1fd093e9-f47c-ffdd-91e9-5039c3ccde61
* Content: [How to publish events that conform to .NET Framework Guidelines - C# Programming Guide](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/events/how-to-publish-events-that-conform-to-net-framework-guidelines)
* Content Source: [docs/csharp/programming-guide/events/how-to-publish-events-that-conform-to-net-framework-guidelines.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/events/how-to-publish-events-that-conform-to-net-framework-guidelines.md)
* Product: **dotnet-csharp**
* Technology: **csharp-fundamentals**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

- Freshen up doc
- Modernize C#
- Add clarification on `EventHandler` naming

Fixes #18624

#### Preview

✔️ [How to publish events that conform to .NET Framework Guidelines](https://review.docs.microsoft.com/en-us/dotnet/csharp/programming-guide/events/how-to-publish-events-that-conform-to-net-framework-guidelines?branch=pr-en-us-18642)
✔️ [How to raise base class events in derived classes](https://review.docs.microsoft.com/en-us/dotnet/csharp/programming-guide/events/how-to-raise-base-class-events-in-derived-classes?branch=pr-en-us-18642)

### CODE REVIEW

File: docs/csharp/programming-guide/events/how-to-publish-events-that-conform-to-net-framework-guidelines.md [hunk 1]
```diff
@@ -1,75 +1,77 @@
...
-        }  
-        private string msg;  
-        public string Message  
-        {  
-            get { return msg; }  
+> The .NET Framework 2.0 introduces a generic version of this delegate, <xref:System.EventHandler%601>. The following examples show how to use both versions.
+
+Although events in classes that you define can be based on any valid delegate type, even delegates that return a value, it is generally recommended that you base your events on the .NET Framework pattern by using <xref:System.EventHandler>, as shown in the following example.
+
+The name `EventHandler` can lead to a bit of confusion, as the delegate itself doesn't actually handle the event but instead is used to raise the event. The <xref:System.EventHandler>, and generic <xref:System.EventHandler%601> are intended to represent a method that will handle an event.
```
Comment (BillWagner) [inline docs/csharp/programming-guide/events/how-to-publish-events-that-conform-to-net-framework-guidelines.md:22]: Consider this language instead:

```suggestion
The name `EventHandler` can lead to a bit of confusion as it doesn't actually handle the event. The <xref:System.EventHandler>, and generic <xref:System.EventHandler%601> are delegate types. A method or lambda expression whose signature matches the delegate definition is the *event handler* and will be invoked when the event is raised.
```

File: docs/csharp/programming-guide/events/how-to-publish-events-that-conform-to-net-framework-guidelines.md [hunk 2]
```diff
@@ -1,75 +1,77 @@
...
-        public event EventHandler<CustomEventArgs> RaiseCustomEvent;  
-        ```  
-  
-## Example  
- The following example demonstrates the previous steps by using a custom EventArgs class and <xref:System.EventHandler%601> as the event type.  
-  
- [!code-csharp[csProgGuideEvents#2](~/samples/snippets/csharp/VS_Snippets_VBCSharp/csProgGuideEvents/CS/Events.cs#2)]  
-  
+
+        public string Message { get; set; }
```
Comment (BillWagner) [inline docs/csharp/programming-guide/events/how-to-publish-events-that-conform-to-net-framework-guidelines.md:36]: I'd make this readonly:

```suggestion
        public string Message { get; }
```

File: samples/snippets/csharp/VS_Snippets_VBCSharp/csProgGuideEvents/CS/Events.cs [hunk 3]
```diff
@@ -180,31 +177,24 @@ static void Main(string[] args)
...
+            Message = message;
         }
-        private string message;
 
-        public string Message
-        {
-            get { return message; }
-            set { message = value; }
-        }
+        public string Message { get; set; }
```
Comment (BillWagner) [inline samples/snippets/csharp/VS_Snippets_VBCSharp/csProgGuideEvents/CS/Events.cs:192]: same comment:

```suggestion
        public string Message { get; }
```

**Review by BillWagner — APPROVED** [review]
This looks great @IEvangelist 

I had one wording suggestion and a couple small code suggestions, and then it's ready to :shipit:

File: docs/csharp/programming-guide/events/how-to-publish-events-that-conform-to-net-framework-guidelines.md (hunk 2 above)
Comment (IEvangelist) [reply docs/csharp/programming-guide/events/how-to-publish-events-that-conform-to-net-framework-guidelines.md:36]: I did initially, and then it broke some code. It is appended to later, see [line 222](https://github.com/dotnet/docs/pull/18642/files#diff-58e13354452c4c47d554013a02f59919R222)

File: samples/snippets/csharp/VS_Snippets_VBCSharp/csProgGuideEvents/CS/Events.cs (hunk 3 above)
Comment (IEvangelist) [reply samples/snippets/csharp/VS_Snippets_VBCSharp/csProgGuideEvents/CS/Events.cs:192]: Same reason as above...

**Pull request merged**